libvast/vast/directory.hpp

```diff
@@ -19,6 +19,12 @@
 
 #ifdef VAST_POSIX
 #  include <dirent.h>
+#else
+
+namespace vast {
+struct DIR;
```

Comment:
Should this really go into the public namespace? It seems like an implementation detail.

Reply:
Can you think of a better alternative? I did't want to put it in the global namespace since the posix `DIR` already lives there when it exists, and I don't even want to think about the ODR can of worms we're going to open when we put this into an unnamed namespace but let `directory` have a `DIR*` member.

Reply:
I would put it at least into namespace `vast::detail`. Namespace `vast` should be as stable as possible.

Alternatively, you can do type erasure and use a void pointer, which you then cast with helper functions in the implementation.

Reply:
But in the posix case, we don't want to have a `detail::DIR*`, we need a `DIR*`. And this is in a header, so we cant just use the namespace. I'm not a fan of using a `void*`, since we *know* the type of the pointer.

If you feel strongly about this, I think the cleanest alternatives would be to either
 - declare a wrapper struct in namespace `detail` and give it a `DIR*` member on posix (ie. one extra indirection)
 - put it in the global namespace after all, or
 - `#define DIR void` in the non-posix case

Reply:
I don't have too strong of an opinion, because our hand-rolled filesystem header will evaporate soon after switching  to `std::filesystem`.

If you're not keen on type erasure using the poor-man's void* pimpling, let's just leave it as. 🙂